### PR TITLE
[WIP] Stop Co-Moving Window After Step

### LINKED ISSUE #1441: Stop the Moving Window
@hightower8083 asked on Gitter about a functionality to stop the moving window after a certain time step.

He currently tries to achieve that with the `Physics_applications/plasma_acceleration/inputs_2d` example, restarting at a certain time step, and then re-starting with a changed input file with `warpx.do_moving_window` changed to `0`, which is not successful (fields are off).

Looks similar to me like:
- https://github.com/ComputationalRadiationPhysics/picongpu/pull/1668
- https://github.com/ComputationalRadiationPhysics/picongpu/pull/2792

@hightower8083 please feel free to add more details and/or the exact modified input scripts you executed, please.

### PR DESCRIPTION
**Pull request by ax3l**
Implement a feature to stop the co-moving window after a certain time step.

Close #1441

## To Do

- [ ] runtime tests cc @hightower8083 
- [ ] add unit test

### CODE REVIEW

**Comment by hightower8083** [comment]
have just pushed a couple of [fixes](https://github.com/ax3l/WarpX/pull/1) -- works for the minimal cases, but will probably need more testing. 
BTW, since we're on it, why not make it even more general adding also the `start_step` ?

File: Source/WarpX.H [hunk 1]
```diff
@@ -462,7 +462,14 @@ public:
...
     static int do_electrostatic;
-    static int do_moving_window;
+    static int do_moving_window; // boolean
+    static int end_moving_window_step; // the last step to move window
+    /** Returns true if the moving window is active for the provided step
+     *
+     * @param step time step
+     * @return true if active, else false
+     */
+    static int moving_window_active(int const step) { return (step < end_moving_window_step) || (end_moving_window_step < 0); }
```
Comment (ax3l) [inline Source/WarpX.H]:
```suggestion
    static int moving_window_active (int const step) {
        bool const step_active = (step < end_moving_window_step) || (end_moving_window_step < 0);
        return do_moving_window && step_active;
    }
```

**Comment by ax3l** [comment]
I was thinking that as well, but since you mentioned this was urgent did not want to over-design.
So you say your specific case already works with this and you can keep going?

**Comment by hightower8083** [comment]
yes, for the case I have all feels more or less OK, but I don't know how boundaries (guards, PMLs) are supposed to react on the window stop. I've only tested a single-GPU config and with an e-beam driver, and after the window stops the PMLs at `zmax` are not absorbing well the field of the beam escaping through this boundary. But this is probably not related to the window stop itself..

Do you have suggestions of how could I proceed with a more in-depth testing? 
also there are some check failing, is that caused by this modification or its something else?

**Comment by ax3l** [comment]
superseded by #2027

**Pull request closed without merging**